packages/jest-diff/src/diff_strings.js

```diff
@@ -25,246 +26,314 @@ type Original = {|
-  chunk.removed ? '-' : chunk.added ? '+' : ' ';
-
-// Given diff character in line of hunk or computed from properties of chunk.
-const getDiffDigit = (char: string): DIFF_DIGIT =>
-  char === '-' ? -1 : char === '+' ? 1 : 0;
-
-// Color for text of line.
-const getColor = (digit: DIFF_DIGIT, onlyIndentationChanged?: boolean) => {
-  if (digit === -1) {
-    return chalk.green; // removed
-  }
-  if (digit === 1) {
-    return chalk.red; // added
-  }
-  return onlyIndentationChanged ? chalk.cyan : chalk.dim;
-};
-
-// Do NOT color leading or trailing spaces if original lines are equal:
-
-// Background color for leading or trailing spaces.
-const getBgColor = (digit: DIFF_DIGIT, onlyIndentationChanged?: boolean) =>
-  digit === 0 && !onlyIndentationChanged ? chalk.bgYellow : chalk.inverse;
+const fgPatchMark = chalk.yellow;
+const fgDelete = chalk.green;
+const fgInsert = chalk.red;
+const fgUnchanged = chalk.dim; // common lines (even indentation same)
+const fgInchanged = chalk.cyan; // common lines (only indentation different)
+const bgUnchanged = chalk.bgYellow; // edge spaces in common line (even indentation same)
+const bgDefault = chalk.inverse; // edge spaces in any other lines
 
 // ONLY trailing if expected value is snapshot or multiline string.
-const highlightTrailingSpaces = (line: string, bgColor: Function): string =>
+const highlightTrailingSpaces = (line: string, bgColor: Chalk): string =>
   line.replace(/\s+$/, bgColor('$&'));
 
 // BOTH leading AND trailing if expected value is data structure.
-const highlightLeadingTrailingSpaces = (
-  line: string,
-  bgColor: Function,
-): string =>
+const highlightLeadingTrailingSpaces = (line: string, bgColor: Chalk): string =>
   // If line consists of ALL spaces: highlight all of them.
   highlightTrailingSpaces(line, bgColor).replace(
     // If line has an ODD length of leading spaces: highlight only the LAST.
     /^(\s\s)*(\s)(?=[^\s])/,
     '$1' + bgColor('$2'),
   );
 
+type Highlight = (line: string, bgColor: Chalk) => string;
+
+const getHighlightSpaces = (bothEdges: boolean): Highlight =>
+  bothEdges ? highlightLeadingTrailingSpaces : highlightTrailingSpaces;
+
 const getAnnotation = (options: ?DiffOptions): string =>
-  chalk.green('- ' + ((options && options.aAnnotation) || 'Expected')) +
+  fgDelete('- ' + ((options && options.aAnnotation) || 'Expected')) +
   '\n' +
-  chalk.red('+ ' + ((options && options.bAnnotation) || 'Received')) +
+  fgInsert('+ ' + ((options && options.bAnnotation) || 'Received')) +
   '\n\n';
 
-// Given string, return array of its lines.
-const splitIntoLines = string => {
-  const lines = string.split('\n');
+type Put = (line: string) => void;
 
-  if (lines.length !== 0 && lines[lines.length - 1] === '') {
-    lines.pop();
-  }
+// Given index interval in expected lines, put formatted delete lines.
+const formatDelete = (
+  aStart: number,
+  aEnd: number,
+  aLinesUn: Array<string>,
+  aLinesIn: Array<string>,
```

Comment:
Can we change the variables names? 
aStart -> start
aLinesUn -> linesUnchanged 
etc 

This comment is valid for all names like these (except when it makes sense  where you have a,b related args)

Reply:
I will review naming and think how to improve comments here to make it less surprising.

The reason for `a` prefix in `formatDelete` and `b` prefix in `formatInsert` is to remind me assumption or convention that took me a while to understand:
* differences in expected value are always formatted as deleted
* differences in received value are always formatted as inserted